### PR TITLE
Added multi-step request creation modal

### PR DESCRIPTION
**Pull request by gretahuang**
`recurrence-request-form.jsx` is under construction; please ignore it. @eric-ls @warrenshen @charlesx2013

### CODE REVIEW

File: app/assets/javascripts/components/locations/confirmation-request-form.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ */
+class ConfirmationForm extends DefaultForm {
+
+  constructor(props) {
+    super(props);
+    this.state = this.props.initData;
+  }
+
+  _prevStep = (e) => {
+    console.log(this.props);
```
Comment (eric-ls) [inline app/assets/javascripts/components/locations/confirmation-request-form.jsx]: remember to remove `console.log`

File: app/assets/javascripts/components/locations/create_request.jsx [hunk 2]
```diff
@@ -2,19 +2,28 @@
...
  * @prop success     - function handler for successful student creation
  */
-class RequestCreationForm extends DefaultForm {
+class RequestCreationModal extends DefaultForm {
 
   constructor(props) {
     super(props);
-    this.state = { location_id : this.props.location_id };
+    this.state = { location_id      : this.props.location_id,
+                   step             : 1,
```
Comment (eric-ls) [inline app/assets/javascripts/components/locations/create_request.jsx]: nit: put the values on a separate line:

``` javascript
this.state = {
    location_id: this.props.location_id,
    ...
}
```

File: app/assets/javascripts/components/locations/create_request.jsx [hunk 3]
```diff
@@ -2,19 +2,28 @@
...
+                 };
   }
 
-  _attemptCreate = (e) => {
+  _attemptCreate = (initData) => {
     const success = (data) => {
       this.props.success();
       this.close();
+      this.setState({ basicForm: {} });
+      this.setState({ step: 1 });
```
Comment (eric-ls) [inline app/assets/javascripts/components/locations/create_request.jsx]: i think you can setState in one method call

``` javascript
this.setState({
    basicForm: {},
    step: 1,
});
```

File: app/assets/javascripts/components/locations/basic-pickup-form.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+/**
+ * @prop initData - saved data associated with the basic portion of the pickup form
+ * @prop nextStep - function handler to move on to next step of pickup creation
+ */
+class BasicPickupForm extends DefaultForm {
+
+  constructor(props) {
+    super(props);
+    this.state = this.props.initData;
+
```
Comment (charlesx2013) [inline app/assets/javascripts/components/locations/basic-pickup-form.jsx]: no space pls

File: app/assets/javascripts/components/locations/recurrence-pickup-form.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    super(props);
+    this.state = {
+      monday    : false,
+      tuesday   : false,
+      wednesday : false,
+      thursday  : false,
+      friday    : false,
+      active    : {},
+    }
+    // let days = ["monday", "tuesday", "wednesday", "thursday", "friday"].map((day)) => {
```
Comment (charlesx2013) [inline app/assets/javascripts/components/locations/recurrence-pickup-form.jsx]: commented code?

File: app/assets/javascripts/components/locations/recurrence-pickup-form.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  _toggleDay = (day) => {
+    let state  = {};
+    state[day] = !this.state[day];
+    this.setState( state );
+  }
+
+  _capitalize = (str) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
```
Comment (charlesx2013) [inline app/assets/javascripts/components/locations/recurrence-pickup-form.jsx]: should be a helper function

Comment (charlesx2013) [inline app/assets/javascripts/components/locations/recurrence-pickup-form.jsx]: as in, make into a separate class.

File: app/assets/javascripts/components/locations/recurrence-pickup-form.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+  _nextStep = (e) => {
+    this.props.nextStep(this.state);
+  }
+
+  _prevStep = (e) => {
+    this.props.prevStep(this.state);
+  }
+
+  // _generateDaySelection => {
```
Comment (charlesx2013) [inline app/assets/javascripts/components/locations/recurrence-pickup-form.jsx]: commented code

**Comment by eric-ls** [comment]
LGTM 👍

**Pull request merged**